package/MDAnalysis/topology/RDKitParser.py

```diff
@@ -0,0 +1,286 @@
+)
+from ..core.topology import Topology
+
+logger = logging.getLogger("MDAnalysis.topology.RDKitParser")
+
+
+class RDKitParser(TopologyReaderBase):
+    """
+    For RDKit structures
+
+    Create the following Attributes:
+     - Atomids
+     - Atomnames
+     - Aromaticities
+     - Elements
+     - Masses
+     - Bonds
+     - Resids
+     - Resnums
+     - Segids
+
+    Guesses the following:
+     - Atomtypes
+
+    Depending on RDKit's input, the following Attributes might be present:
+     - Charges
+     - Resnames
+     - AltLocs
+     - ChainIDs
+     - Occupancies
+     - Tempfactors
+     
+    """
+    format = 'RDKIT'
+
+    @staticmethod
+    def _format_hint(thing):
+        """Can this Parser read object *thing*?
+
+        .. versionadded:: 1.0.0
+        """
+        try:
+            from rdkit import Chem
+        except ImportError:  # if no rdkit, probably not rdkit
+            return False
+        else:
+            return isinstance(thing, Chem.Mol)
+
+    def parse(self, **kwargs):
+        """Parse RDKit into Topology
+
+        Returns
+        -------
+        MDAnalysis Topology object
+        """
+
+        mol = self.filename
+
+        # Atoms
+        names = []
+        resnums = []
+        resnames = []
+        elements = []
+        masses = []
+        charges = []
+        aromatics = []
+        ids = []
+        atomtypes = []
+        segids = []
+        altlocs = []
+        chainids = []
+        occupancies = []
+        tempfactors = []
+
+        # check if multiple charges present
+        atom = mol.GetAtomWithIdx(0)
+        if atom.HasProp('_GasteigerCharge') and (
+        atom.HasProp('_TriposPartialCharge')
+        ):
+            warnings.warn(
+                'Both _GasteigerCharge and _TriposPartialCharge properties '
+                'are present. Using Gasteiger charges by default.')
+
+        for atom in mol.GetAtoms():
+            ids.append(atom.GetIdx())
+            elements.append(atom.GetSymbol())
+            masses.append(atom.GetMass())
+            aromatics.append(atom.GetIsAromatic())
+            mi = atom.GetMonomerInfo()
+            if mi: # atom name and residue info are present
+                names.append(mi.GetName().strip())
+                resnums.append(mi.GetResidueNumber())
+                resnames.append(mi.GetResidueName())
+                segids.append(mi.GetSegmentNumber())
+                altlocs.append(mi.GetAltLoc().strip())
+                chainids.append(mi.GetChainId().strip())
+                occupancies.append(mi.GetOccupancy())
+                tempfactors.append(mi.GetTempFactor())
+            else:
+                # atom name (MOL2 only)
+                try:
+                    names.append(atom.GetProp('_TriposAtomName'))
+                except KeyError:
+                    pass
+                # atom type (MOL2 only)
+                try:
+                    atomtypes.append(atom.GetProp('_TriposAtomType'))
```

Comment:
I don't think "guaranteed" and "MOL2" can be used in the same sentence unfortunately, especially when it comes to atom types. Nevertheless, I made a quick test with the `Chem.MolFromMol2Block`: neither the atom name nor atom type can be blank, otherwise the resulting object is just `None`. So I guess if reading from a MOL2 file, they both have to be present otherwise RDKit will fail.

Reply:
> I don't think "guaranteed" and "MOL2" can be used in the same sentence unfortunately,

😆 That's a very good point.

> So I guess if reading from a MOL2 file, they both have to be present otherwise RDKit will fail.

If you think it's worth it, it might make sense to put them within the same try/except block then , although maybe with a comment stating that we are making that assumption. My main concern here was whether or not there would be a large performance penalty to triggering the except twice if you were passing a non-MOL2 file, maybe worth benchmarking?

Reply:
Tried with the SMILES string of caffeine (24 atoms, without generating coordinates), 10 runs, 100 loops each:
* same block: 8.51 ms ± 363 µs per loop
* 2 try/except blocks: 11.3 ms ± 1.07 ms per loop

With PDB_helix file (137 atoms):
* same block: 7.73 ms ± 255 µs per loop
* 2 try/except blocks: 7.31 ms ± 472 µs per loop

With MOL2 file (49 atoms):
* same block: 13.9 ms ± 458 µs per loop
* 2 try/except blocks: 14.7 ms ± 648 µs per loop

I don't see a large performance drop there so I guess we can keep it at 2 ?

Reply:
> I don't see a large performance drop there so I guess we can keep it at 2 ?

I think that's fair unless @fiona-naughton or @richardjgowers have a different opinion here.

Reply:
This looks fine, we can do a performance pass if it ever becomes notably slow. I’m imagining any chem perception RDKit does is magnitudes slower than try/excepts on our end.